### PR TITLE
[#44] Handle Cabal packages without the `hs-source-dirs` field

### LINKED ISSUE #44: Handle Cabal packages without the `hs-source-dirs` field
Some Haskell packages don't have the `hs-source-dirs` field in their `.cabal` files. In such packages, modules are placed directly on top-level. One example of such package:

* https://github.com/haskell/base64-bytestring/blob/master/base64-bytestring.cabal

So instead of `src/Data/ByteString/Base64.hs` file, you have simply `Data/ByteString/Base64.hs`. Current implementation of `parseCabalExtensions` will return empty map in such cases. But I guess it needs to be patched. I see packages with such structure from time to time in the Haskell ecosystem.

### PR DESCRIPTION
**Pull request by chshersh**
Resolves #44

### CODE REVIEW

File: src/Extensions/Cabal.hs [hunk 1]
```diff
@@ -14,12 +14,16 @@ module Extensions.Cabal
     , parseCabalFileExtensions
     , parseCabalExtensions
     , extractCabalExtensions
+
+      -- * Bridge between Cabal and GHC extensions
+    , cabalToGhcExtension
+    , toGhcExtension
```
Comment (chshersh) [inline src/Extensions/Cabal.hs:20]: Decided to reexports those functions, so others won't need to write them by themselves anymore

Comment (vrom911) [reply src/Extensions/Cabal.hs:20]: Very kind of you :fishing_pole_and_fish: :hugs:

File: src/Extensions/Parser.hs [hunk 2]
```diff
@@ -166,7 +166,11 @@ commentP = newLines *> (try singleLineCommentP <|> try multiLineCommentP) <* new
         (string "{-" *> manyTill anyChar (try $ string "-}"))
 
 cppP :: Parser [a]
-cppP = [] <$ many newline <* try (char '#' <* noneOf "-") <* manyTill anyChar (try endOfLine)
+cppP =
+    [] <$ many newline
+       <* try (char '#' <* noneOf "-")
+       <* manyTill anyChar (try endOfLine)
+       <* newLines
```
Comment (vrom911) [inline src/Extensions/Parser.hs:173]: :golf:

**Review by vrom911 — APPROVED** [review]
Really nice! :clapper:

**Pull request merged**